pepper-apis/src/main/java/org/broadinstitute/ddp/model/study/PasswordPolicy.java

```diff
@@ -0,0 +1,129 @@
+package org.broadinstitute.ddp.model.study;
+
+import java.util.Set;
+
+import com.google.gson.annotations.SerializedName;
+import org.broadinstitute.ddp.exception.DDPException;
+
+public class PasswordPolicy {
+
+    public static final int DEFAULT_MAX_REPEATED_CHARS = 2;
+    public static final int DEFAULT_MIN_CHAR_CLASSES = 3;
+    public static final int MAX_PASSWORD_LENGTH = 128;
+
+    private transient PolicyType type;
+    @SerializedName("minLength")
+    private int minLength;
+    @SerializedName("minCharClasses")
+    private int minCharClasses;
+    @SerializedName("charClasses")
+    private Set<CharClass> charClasses;
+    @SerializedName("maxRepeatedChars")
+    private Integer maxRepeatedChars;
+
+    public static PasswordPolicy fromType(PolicyType type, Integer minLength) {
+        switch (type) {
+            case NONE:
+                return none(minLength);
+            case LOW:
+                return low(minLength);
+            case FAIR:
+                return fair(minLength);
+            case GOOD:
+                return good(minLength);
+            case EXCELLENT:
+                return excellent(minLength);
+            default:
+                throw new DDPException("Unhandled password policy type " + type);
+        }
+    }
+
+    public static PasswordPolicy none(Integer minLength) {
+        minLength = (minLength == null ? PolicyType.NONE.getDefaultMinPasswordLength() : minLength);
```

Comment:
What if we did something like `Optional.ofNullable(minLength).orElse(PolicyType.NONE.getDefaultMinPasswordLength())` here? It minimizes some of the in-line repetition, and the intent feels more readable.

Reply:
Either seems fine. I don't think ternary operator will diminish the intent since it's common usage in Java. Also, might be micro-optimizing since I don't know how Java compiles it, but `Optional` approach involves creating a new object and a method call whereas ternary is more of a "native" operator in Java. Maybe I can flip the logic so it reads:
```java
min != null ? min : getDefault()
```

Reply:
It does create an object, but I don't think we are anywhere _near_ the point where optimizations like that are going to make a significant difference in the response time of the server. As for the intent, the `Optional` suggestion was primarily to reduce the in-line repetition of `minLength`- I'm very intimately familiar with the ternary operation in a number of languages.